eth/downloader/downloader_test.go

```diff
@@ -983,24 +730,27 @@ func checkProgress(t *testing.T, d *Downloader, stage string, want ethereum.Sync
 }
 
 // Tests that synchronisation progress (origin block number and highest block
-// number) is tracked and updated correctly in case of a fork (or manual head
-// revertal).
+// number) is tracked and updated correctly in case of manual head reversion
 func TestForkedSyncProgress68Full(t *testing.T)  { testForkedSyncProgress(t, eth.ETH68, FullSync) }
```

Comment:
I've modified this to test the following behavior:
* node is requested to sync to chain A, syncs to chain A
* nodes is requested to sync to fork B:
```
A - A_0, A_1, A_2, ..., A_n
B - A_0, A_1, A_2, ... A_(x-n), B_(x-n+1), ..., B_n
```

Test case will check that the sync origin after syncing `B` is at the fork block (`B_(x-n+1)`).

However, I'm having trouble getting it to sync `B`.  Through debugging, I see that the state/chain backfilling is kicked off by a successful skeleton sync and fails [here](https://github.com/ethereum/go-ethereum/blob/822e60b93f85c1e07f3b875208c04faf8d668ead/eth/downloader/downloader.go#L462).  I'm working on debugging this.